### PR TITLE
feat(json-reporter): Add ability to generate mutation report in json format

### LINKED ISSUE #225: Reporter setting
Now, the default reporters are: progress reporter and report reporter. To show the progress and the end result.

On build servers the progress reporter is not useful. Users could want to change the reporter to `ConsoleDotRepoter` and no report for example. To give users full freedom they can mix and match the reporters themselves. This will mean a change for the current `reporter` setting, which takes only one value. It should take multiple values, and the name should be plural.

Example:
`--reporters ['dots']`
`--reporters ['progress', 'clear-text']`

All options will be:
`['clear-text', 'progress', 'dots']`

The default should be:
`['progress', 'clear-text']`

### LINKED ISSUE #277: Make it possible to have NoValue arguments passed to Stryker for boolean operations
https://github.com/stryker-mutator/stryker-net/blob/2f6f459a6d74a178c31ab0e3f62850a8912d1c9c/src/Stryker.CLI/Stryker.CLI/CLIOptions.cs#L43

https://github.com/stryker-mutator/stryker-net/blob/2f6f459a6d74a178c31ab0e3f62850a8912d1c9c/src/Stryker.CLI/Stryker.CLI/StrykerCLI.cs#L139

 I believe if we remove the <enable> from the shorthand and add the option at command line argument registration to set it to NoValue instead of the hardcoded SingleValue, passing -dm should be enough for a true value, and passing true on the commandline is no longer required.

### PR DESCRIPTION
**Pull request by Mobrockers**
Closes first step of #109 
Closes #277 
Closes #225

### CODE REVIEW

File: src/Stryker.Core/Stryker.Core/Stryker.Core.csproj [hunk 1]
```diff
@@ -17,6 +17,7 @@
     <PackageLicenseUrl>https://github.com/stryker-mutator/stryker-net/blob/master/LICENSE</PackageLicenseUrl>
     <PackageIconUrl>https://raw.githubusercontent.com/stryker-mutator/stryker/master/stryker-80x80.png</PackageIconUrl>
     <RepositoryUrl>https://github.com/stryker-mutator/stryker-net</RepositoryUrl>
+    <Configurations>Debug;Release</Configurations>
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core/Stryker.Core.csproj]: Is this needed?

File: src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs [hunk 2]
```diff
@@ -54,15 +53,35 @@ public class StrykerOptions
             ThresholdOptions = ValidateThresholds(thresholdHigh, thresholdLow, thresholdBreak);
         }
 
-        private string ValidateReporter(string reporter)
+        private IEnumerable<Reporter> ValidateReporters(string[] reporters)
         {
-            if (reporter != "Console" && reporter != "ReportOnly")
+            if (reporters == null)
```
[the remark this answers is not in the document]
Comment (Mobrockers) [reply src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs]: Imo we should give people the option to disable all reporters. Passing in an empty array will do that. I'll add an explicit yield break at the end to make sure we return an empty IEnumerable if no reporters are enabled.

File: src/Stryker.Core/Stryker.Core/Reporters/JsonReporter.cs [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+            public IList<JsonMutant> Mutants { get; } = new List<JsonMutant>();
+
+            public IList<JsonReportComponent> ChildResults { get; } = new List<JsonReportComponent>();
+
+            public static JsonReportComponent FromProjectComponent(IReadOnlyInputComponent component, ThresholdOptions thresholdOptions)
+            {
+                var report = new JsonReportComponent
+                {
+                    Name = component.Name,
+                    TotalMutants = component.ReadOnlyMutants.Count(),
```
[the remark this answers is not in the document]
Comment (Mobrockers) [reply src/Stryker.Core/Stryker.Core/Reporters/JsonReporter.cs]: But we're listing all of those values in the json as well, so when you add up all the numbers we mention should they not add up to TotalMutants?

[the remark this answers is not in the document]
Comment (Mobrockers) [reply src/Stryker.Core/Stryker.Core/Reporters/JsonReporter.cs]: I have added the property PossibleMutants instead. This property contains a count of the mutants that were possible to apply, including the ones skipped by users, but excluding the ones that could not compile

**Pull request merged**